tests/run/i7000.scala

```diff
@@ -0,0 +1,16 @@
+import scala.deriving._
+import compiletime._
+
+object Test extends App {
+  case class B(v: Double) extends AnyVal
+  val m0 = the[Mirror.ProductOf[B]]
+
+  val v0 = m0.fromProduct(Tuple1(23.0))
+  assert(v0 == B(23.0))
+
+  case class C[T](v: T) extends AnyVal
```

Comment:
What happens for value classes which don't extend product (so don't have `case`) ?

Reply:
Currently only case classes (value or otherwise) have product mirrors generated automatically, so that case doesn't arise. Someone could provide a mirror for a non-case value class, and then they'd have to deal with the bridge collision themselves.